### PR TITLE
32687 Provider selection information missing for CC request

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
This PR fixes issue with provider information not displaying on the review page.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#32687

## Testing done
Unit and e2e testing

## Acceptance criteria
- [ ] All test should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by cferris32 — APPROVED** [review]
This is looking good to me 👍🏼

File: src/applications/vaos/new-appointment/components/ReviewPage/CommunityCareSection.jsx [hunk 1]
```diff
@@ -4,20 +4,24 @@ import ReasonForAppointmentSection from './ReasonForAppointmentSection';
...
 import PreferredProviderSection from './PreferredProviderSection';
 import SelectedProviderSection from './SelectedProviderSection';
+import { useSelector } from 'react-redux';
+import { selectFeatureCCIterations } from '../../../redux/selectors';
+
+export default function CommunityCareSection({ data, vaCityState }) {
+  const provider = data.communityCareProvider;
+  const hasProvider =
+    !!provider && !!Object.keys(data.communityCareProvider).length;
+  const myData = { ...data, hasCommunityCareProvider: hasProvider };
```
Comment (vbahinwillit) [inline src/applications/vaos/new-appointment/components/ReviewPage/CommunityCareSection.jsx:14]: Adding 'hasCoummunityCareProvider' flag to data object to be used with PreferredProviderSection. This flag is used to determine whether provider information is display or not.

File: src/applications/vaos/new-appointment/components/ReviewPage/CommunityCareSection.jsx [hunk 2]
```diff
@@ -4,20 +4,24 @@ import ReasonForAppointmentSection from './ReasonForAppointmentSection';
...
     <>
       <PreferredDatesSection data={data} />
       <hr aria-hidden="true" className="vads-u-margin-y--2" />
-      {!hasResidentialAddress && (
-        <PreferredProviderSection data={data} vaCityState={vaCityState} />
+      {!featureCCIteration && (
+        <PreferredProviderSection data={myData} vaCityState={vaCityState} />
       )}
-      {hasResidentialAddress && (
+      {featureCCIteration && (
```
Comment (jenniemc) [inline src/applications/vaos/new-appointment/components/ReviewPage/CommunityCareSection.jsx:24]: How did you know to substitute `hasResidentialAddress` with `featureCCIteration`?

File: src/applications/vaos/new-appointment/index.jsx [hunk 3]
```diff
@@ -113,19 +111,18 @@ export function NewAppointment() {
...
           />
         )}
-        {!featureCCIteration &&
-          !hasResidentialAddress && (
-            <Route
-              path={`${match.url}/community-care-preferences`}
-              component={CommunityCarePreferencesPage}
-            />
-          )}
+        {!featureCCIteration && (
```
Comment (jenniemc) [inline src/applications/vaos/new-appointment/index.jsx:120]: why did you remove the `hasResidentialAddress` condition?

File: src/applications/vaos/new-appointment/index.jsx [hunk 4]
```diff
@@ -113,19 +111,18 @@ export function NewAppointment() {
           path={`${match.url}/how-to-schedule`}
           component={ScheduleCernerPage}
         />
-        {(featureCCIteration || hasResidentialAddress) && (
+        {featureCCIteration && (
```
Comment (jenniemc) [inline src/applications/vaos/new-appointment/index.jsx:114]: why did you remove the `hasResidentialAddress`?

**Review by jenniemc — APPROVED** [review]
This is looking good! I just had a few questions as to how you debug to find the solution especially about replacing `hasResidentialAddress` for `featureCCIteration`.

File: src/applications/vaos/new-appointment/components/ReviewPage/CommunityCareSection.jsx (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/new-appointment/components/ReviewPage/CommunityCareSection.jsx:24]: Great question!. I had the same concern so I had a conversation with Lauren about this and I don't think the having an address condition was ever correct. So, I going to try this fix and hopefully I have not broken anything else!
